.vscode/c_cpp_properties.json

```diff
@@ -0,0 +1,15 @@
+{
+    "configurations": [
+        {
+            "name": "Linux",
+            "includePath": [
+                "${workspaceFolder}/**"
+            ],
+            "defines": [],
+            "compilerPath": "/usr/bin/gcc",
+            "cppStandard": "c++11",
+            "intelliSenseMode": "clang-x64"
+        }
+    ],
+    "version": 4
```

Comment:
What does this version mean? 

In general, can you explain what these json files do? For example, why is `intelliSenseMode` using clang but the `compilerPath` points to gcc? What are these properties used for?

Reply:
this is recommended settings I got from MS: https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference

Not really sure what is each setting...